acvm-repo/brillig_vm/src/arithmetic.rs

```diff
@@ -107,89 +112,71 @@ pub(crate) fn evaluate_binary_int_op<F: AcirField>(
     })
 }
 
-fn evaluate_binary_int_op_128(
+fn evaluate_binary_int_op_u1(
     op: &BinaryIntOp,
-    lhs: u128,
-    rhs: u128,
-) -> Result<u128, BrilligArithmeticError> {
+    lhs: bool,
+    rhs: bool,
+) -> Result<bool, BrilligArithmeticError> {
     let result = match op {
-        BinaryIntOp::Add => lhs.wrapping_add(rhs),
-        BinaryIntOp::Sub => lhs.wrapping_sub(rhs),
-        BinaryIntOp::Mul => lhs.wrapping_mul(rhs),
+        BinaryIntOp::Add | BinaryIntOp::Sub => lhs ^ rhs,
+        BinaryIntOp::Mul => lhs & rhs,
         BinaryIntOp::Div => {
-            if rhs == 0 {
+            if !rhs {
                 return Err(BrilligArithmeticError::DivisionByZero);
             } else {
-                lhs / rhs
+                lhs
             }
         }
-        BinaryIntOp::Equals => (lhs == rhs) as u128,
-        BinaryIntOp::LessThan => (lhs < rhs) as u128,
-        BinaryIntOp::LessThanEquals => (lhs <= rhs) as u128,
+        BinaryIntOp::Equals => lhs == rhs,
+        BinaryIntOp::LessThan => !lhs & rhs,
+        BinaryIntOp::LessThanEquals => lhs <= rhs,
```

Comment:
Could use the bitwise definition for `<=`:
```
lhs <= rhs = !(lhs & !rhs) = !lhs | !!rhs = !lhs | rhs
```
But I'd prefer to leave `!lhs & rhs` as `lhs < rhs` to make it easier to read.

Reply:
:shrug: I'm just following clippy on this.